src/main.ts

```diff
@@ -1,8 +1,15 @@
 import { NestFactory } from '@nestjs/core';
 import { AppModule } from './app.module';
+import * as dotenv from 'dotenv';
+
+dotenv.config();
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
-  await app.listen(3000);
+  const port = process.env.PORT || 3000;
+  await app.listen(port, '127.0.0.1');
+  const url = await app.getUrl();
+  const displayUrl = url.replace('127.0.0.1', 'localhost');
```

Comment:
What is the reason that you need this line `const displayUrl = url.replace('127.0.0.1', 'localhost');`